src/utils/editor/source-search.js

```diff
@@ -152,11 +152,12 @@ function doSearch(ctx, rev, query, keepSelection, modifiers: SearchModifiers) {
 
     updateOverlay(cm, state, query, modifiers);
     updateCursor(cm, state, keepSelection);
-    searchNext(ctx, rev, query, newQuery, modifiers);
+    let searchLocation = searchNext(ctx, rev, query, newQuery, modifiers);
 
+    matchIndex = searchLocation ? searchLocation.from : null;
     // NOTE: We would like to find the correct match index based on where the
     // match is in the document.
-    state.matchIndex = matchIndex;
+    state.matchIndex = matchIndex ? matchIndex.ch : -1;
```

Comment:
I noticed that state.matchIndex is not used anywhere in the codebase... we are using the return value of doSearch instead. Should I remove it?